src/goto-programs/goto_trace.cpp

```diff
@@ -143,6 +143,12 @@ std::string trace_value_binary(
     {
       return expr.is_true()?"1":"0";
     }
+    else if(type.id()==ID_integer)
+    {
+      mp_integer i;
+      if(!to_integer(expr, i) && i>=0)
```

Comment:
Why is it necessary that the result is non-negative?

Reply:
Otherwise it prints a negative binary number, which is at least unusual. Conversion to two's complement (as done for signed int) isn't obvious, as there is no width.

Reply:
Ok.